### PR TITLE
Allow codeception 5

### PR DESCRIPTION
**Pull request by akalineskou**
These were the changes I had to do to make this work with codeception 5.

I've ran the whole test suite in a project (private) and had no issues.

- [x] needs v5 release of codeception https://github.com/Codeception/Codeception/releases
- [x] Update compoer.json codeception to `^5.0`

### CODE REVIEW

**Comment by akalineskou** [comment]
Codeception v5 does not support php <8 https://github.com/Codeception/Codeception/blob/5.0.0-RC3/composer.json
So I removed it from compose.json and the tests

Will have to wait for v5 to be released, since as of writing this the latest version is RC3 https://github.com/Codeception/Codeception/releases

We can run the tests with 5.0.0-RC3 in composer.json to see if they pass (and it will be updated when v5 is released)

**Comment by wazum** [comment]
Codeception 5 was released today
https://github.com/Codeception/Codeception/releases

**Comment by akalineskou** [comment]
Updated composer with codeception v5
Also added to allow v3 of codeception/module-asserts https://github.com/Codeception/module-asserts/releases

File: composer.json [hunk 1]
```diff
@@ -13,10 +13,10 @@
         }
     ],
     "require": {
-        "php": "^7.4 || ^8.0",
+        "php": "^8.0",
```
Comment (Jamesking56) [inline composer.json:16]: Again here, keep allowing PHP 7.4

File: composer.json [hunk 2]
```diff
@@ -13,10 +13,10 @@
...
         "ext-json": "*",
         "guzzlehttp/guzzle": "^6.0 || ^7.0",
-        "codeception/codeception": "^3.0 || ^4.0",
+        "codeception/codeception": "^5.0",
```
Comment (Jamesking56) [inline composer.json:19]: Probably drop v3 but keep v4 and v5

**Comment by Jamesking56** [comment]
Test failure @akalineskou

File: src/Module/MailCatcher.php [hunk 3]
```diff
@@ -9,20 +9,17 @@
 
 class MailCatcher extends Module
```
Comment (Jamesking56) [inline src/Module/MailCatcher.php:10]: Potentially could `declare(strict_types=1);` for this file now

**Comment by Jamesking56** [comment]
Hit a bit of a problem here in trying to support both Codeception v4 and v5 in `Module` class:

In Codeception v4, `$config` has no type:

https://github.com/Codeception/Codeception/blob/18beff367306d66f955add5fa74e590a5210e219/src/Codeception/Module.php#L57

However, in v5 it does:

https://github.com/Codeception/Codeception/blob/667dfa4acde020b4328aa5570818cc8cb305f8c4/src/Codeception/Module.php#L49

We need to extend `$config`, so how do we do so without causing a PHP fatal error due to not matching the typing of the parent?

**Comment by akalineskou** [comment]
You could release version 3 which would not support codeception 4... I mean at some point people will have to upgrade...

**Comment by schmunk42** [comment]
> You could release version 3 which would not support codeception 4... I mean at some point people will have to upgrade...

@Jamesking56 Please do that :)

**Comment by Jamesking56** [comment]
Codeception v5 is now supported in [v3.0.0](https://github.com/captbaritone/codeception-mailcatcher-module/releases/tag/3.0.0)

**Pull request closed without merging**